SimDataFormats/Track/interface/SimTrack.h

```diff
@@ -34,8 +34,9 @@ class SimTrack : public CoreSimTrack {
   bool noVertex() const { return ivert == -1; }
 
   /// index of the corresponding Generator particle in the Event container (-1 if no Genpart)
-  int genpartIndex() const { return igenpart; }
-  bool noGenpart() const { return igenpart == -1; }
+  bool isPrimary() const { return (trackInfo_ >> 1) & 1; }
```

Comment:
@AuroraPerego , track info defined in all cases, when tracking of given particle is started.